tests/ui/lifetimes/issue-79187-2.stderr

```diff
@@ -43,6 +43,10 @@ note: the lifetime requirement is introduced here
    |
 LL | fn take_foo(_: impl Foo) {}
    |                     ^^^
+help: consider specifying the type of the closure parameters
+   |
+LL |     take_foo(|a: &_| a);
```

Comment:
Is the `ty_span` assigned wrongly here? I believe it should be assigned to `impl Foo` instead of `a`. I can open another PR to fix this if this looks like a bug.

Reply:
... so that we can produce diagnostics message like `a: &impl Foo`

Reply:
The span may be messed up because of the desugaring into a parameter :shrug: